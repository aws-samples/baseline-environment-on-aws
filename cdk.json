{
  "app": "npx ts-node bin/able-base-sa.ts",
  "context": {
    "dev": {
      "description": "Environment variables for able-guest-*-samples.ts",
      "envName": "Development",
      "vpcCidr": "10.100.0.0/16",
      "securityNotifyEmail": "notify-security@example.com",
      "monitoringNotifyEmail": "notify-monitoring@example.com",
      "dbUser": "dbadmin",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX",
        "channelIdMon": "C01YYYYYYYY"
      }
    },
    "stage": {
      "description": "Environment variables for able-guest-*-samples.ts",
      "envName": "Staging",
      "vpcCidr": "10.110.0.0/16",
      "securityNotifyEmail": "notify-security@example.com",
      "monitoringNotifyEmail": "notify-monitoring@example.com",
      "dbUser": "dbadmin",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX",
        "channelIdMon": "C01YYYYYYYY"
      }
    },
    "prod": {
      "description": "Environment variables for able-guest-*-samples.ts",
      "envName": "Production",
      "vpcCidr": "10.120.0.0/16",
      "securityNotifyEmail": "notify-security@example.com",
      "monitoringNotifyEmail": "notify-monitoring@example.com",
      "dbUser": "dbadmin",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX",
        "channelIdMon": "C01YYYYYYYY"
      }
    },
    "ctaudit": {
      "description": "Environment variables for able-base-ct-audit.ts",
      "env": {
        "account": "123456789012",
        "region": "ap-northeast-1"
      },
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdAgg": "C01ZZZZZZZZ"
      }
    }
  }
}
