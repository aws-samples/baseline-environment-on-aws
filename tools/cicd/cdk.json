{
  "app": "npx ts-node --prefer-ts-exts bin/bleadeploy.ts",
  "context": {
    "dev": {
      "description": "Context samples for Dev - Anonymous account & region",
      "envName": "Development",
      "githubRepositoryOwner": "ownername",
      "githubRepositoryName": "repositoryname",
      "githubTargetBranch": "staging",
      "codestarConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    },
    "stage": {
      "description": "Context samples for Staging - Specific account & region  ",
      "env": {
        "account": "111111111111",
        "region": "ap-northeast-1"
      },
      "envName": "Staging",
      "githubRepositoryOwner": "ownername",
      "githubRepositoryName": "repositoryname",
      "githubTargetBranch": "staging",
      "codestarConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    },
    "prod": {
      "description": "Context samples for Production - Specific account & region  ",
      "env": {
        "account": "222222222222",
        "region": "ap-northeast-1"
      },
      "envName": "Production",
      "githubRepositoryOwner": "ownername",
      "githubRepositoryName": "repositoryname",
      "githubTargetBranch": "main",
      "codestarConnectionArn": "arn:aws:codestar-connections:ap-northeast-1:xxxxxxxxxxxx:connection/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    }
  }
}
