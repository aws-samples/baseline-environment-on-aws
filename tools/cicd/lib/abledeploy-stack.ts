import * as cdk from "@aws-cdk/core";
import * as iam from "@aws-cdk/aws-iam";
import * as codebuild from "@aws-cdk/aws-codebuild";

export interface ABLEdeployStackProps extends cdk.StackProps {
  githubRepositoryOwner: string;
  githubRepositoryName: string;
  githubTargetBranch: string;
}

export class ABLEdeployStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props: ABLEdeployStackProps) {
    super(scope, id, props);

    // Create GitHub source for CodeBuild
    const gitHubSource = codebuild.Source.gitHub({
      owner: props.githubRepositoryOwner,
      repo: props.githubRepositoryName,
      webhook: true,
      fetchSubmodules: true,
      webhookFilters: [
        codebuild.FilterGroup.inEventOf(codebuild.EventAction.PUSH).andBranchIs(
          props.githubTargetBranch
        ),
      ],
    });

    // Create CodeBuild from GitHub source
    const project = new codebuild.Project(this, "ABLEdeployProject", {
      source: gitHubSource,
    });

    // Change role for CodeBuild to attache Admin access for using CDK
    const role = project.role;
    if (role) {
      role.addManagedPolicy(
        iam.ManagedPolicy.fromAwsManagedPolicyName("AdministratorAccess")
      );
    }
  }
}
