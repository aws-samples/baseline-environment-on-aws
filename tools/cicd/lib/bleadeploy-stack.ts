import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { aws_iam as iam } from 'aws-cdk-lib';
import { aws_codebuild as codebuild } from 'aws-cdk-lib';
import { aws_codepipeline as codepipeline } from 'aws-cdk-lib';
import { aws_codepipeline_actions as codepipeline_actions } from 'aws-cdk-lib';

export interface BLEAdeployStackProps extends cdk.StackProps {
  githubRepositoryOwner: string;
  githubRepositoryName: string;
  githubTargetBranch: string;
  codestarConnectionArn: string;
}

export class BLEAdeployStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: BLEAdeployStackProps) {
    super(scope, id, props);

    const pipeline = new codepipeline.Pipeline(this, `${id}-Pipeline`, {
      pipelineName: `${id}-Pipeline`,
    });

    // You just have to select GitHub as the source when creating the connection in the console
    const sourceOutput = new codepipeline.Artifact();
    const sourceAction = new codepipeline_actions.CodeStarConnectionsSourceAction({
      actionName: `${id}-GitHubSource`,
      owner: props.githubRepositoryOwner,
      repo: props.githubRepositoryName,
      branch: props.githubTargetBranch,
      connectionArn: props.codestarConnectionArn,
      output: sourceOutput,
    });

    const deployRole = new iam.Role(this, `${id}-CodeBuildDeployRole`, {
      assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
      managedPolicies: [
        {
          managedPolicyArn: 'arn:aws:iam::aws:policy/AdministratorAccess',
        },
      ],
    });

    const applicationBuild = new codebuild.PipelineProject(this, `${id}-CodeBuildProject`, {
      projectName: `${id}-CodeBuildProject`,
      role: deployRole,
      environment: {
        buildImage: codebuild.LinuxBuildImage.AMAZON_LINUX_2_3,
        environmentVariables: {
          AWS_DEFAULT_REGION: {
            type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
            value: this.region,
          },
        },
      },
    });

    const buildAction = new codepipeline_actions.CodeBuildAction({
      actionName: `${id}-BuildAction`,
      project: applicationBuild,
      input: sourceOutput,
      runOrder: 3,
    });

    pipeline.addStage({
      stageName: 'Source',
      actions: [sourceAction],
    });
    pipeline.addStage({
      stageName: 'Build',
      actions: [buildAction],
    });
  }
}
