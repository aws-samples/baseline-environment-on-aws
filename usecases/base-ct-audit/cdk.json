{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-ct-audit.ts",
  "context": {
    "dev-audit": {
      "description": "Context samples for ControlTower Audit Account - Specific account & region",
      "env": {
        "account": "333333333333",
        "region": "ap-northeast-1"
      },
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdAgg": "C01ZZZZZZZZ"
      }
    }
  }
}
