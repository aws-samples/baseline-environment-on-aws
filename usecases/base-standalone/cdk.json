{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-sa.ts",
  "context": {
    "dev": {
      "description": "Context samples for Dev - Anonymous account & region",
      "envName": "Development",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C00XXXXXXXX"
      }
    },
    "stage": {
      "description": "Context samples for Staging - Specific account & region  ",
      "env": {
        "account": "111111111111",
        "region": "ap-northeast-1"
      },
      "envName": "Staging",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX"
      }
    },
    "prod": {
      "description": "Context samples for Production - Specific account & region  ",
      "env": {
        "account": "222222222222",
        "region": "ap-northeast-1"
      },
      "envName": "Production",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C02XXXXXXXX"
      }
    }
  }
}
