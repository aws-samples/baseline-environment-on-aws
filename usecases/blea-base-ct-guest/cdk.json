{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-ct-guest.ts"
}
