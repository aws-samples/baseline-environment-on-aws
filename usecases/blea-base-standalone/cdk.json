{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-standalone.ts"
}
