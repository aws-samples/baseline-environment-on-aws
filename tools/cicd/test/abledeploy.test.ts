import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as AblEdeploy from '../lib/abledeploy-stack';

// Load cdk.json to get context parameters
import * as cdk_json from '../cdk.json';

const procEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const pjPrefix = 'ABLE';
const app = new cdk.App();
const envKey = 'dev';
const envVals = cdk_json['context'][envKey];

describe(`${pjPrefix} Stacks`, () => {
  test('Empty Stack', () => {
    const deployStack = new AblEdeploy.ABLEdeployStack(app, 'DeployStack', {
      githubRepositoryOwner: 'githubRepositoryOwner',
      githubRepositoryName: 'githubRepositoryName',
      githubTargetBranch: 'githubTargetBranch',
      env: procEnv,
    });

    // test with snapshot
    expect(SynthUtils.toCloudFormation(deployStack)).toMatchSnapshot();
  });
});
