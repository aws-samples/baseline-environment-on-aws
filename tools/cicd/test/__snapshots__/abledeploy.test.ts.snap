// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ABLE Stacks Empty Stack 1`] = `
Object {
  "Resources": Object {
    "ABLEdeployProject345899AE": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "NO_ARTIFACTS",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:1.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "ABLEdeployProjectRole33EB96A4",
            "Arn",
          ],
        },
        "Source": Object {
          "GitSubmodulesConfig": Object {
            "FetchSubmodules": true,
          },
          "Location": "https://github.com/githubRepositoryOwner/githubRepositoryName.git",
          "ReportBuildStatus": true,
          "Type": "GITHUB",
        },
        "Triggers": Object {
          "FilterGroups": Array [
            Array [
              Object {
                "Pattern": "PUSH",
                "Type": "EVENT",
              },
              Object {
                "Pattern": "refs/heads/githubTargetBranch",
                "Type": "HEAD_REF",
              },
            ],
          ],
          "Webhook": true,
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "ABLEdeployProjectRole33EB96A4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AdministratorAccess",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ABLEdeployProjectRoleDefaultPolicyF0179D0A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "ABLEdeployProject345899AE",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "ABLEdeployProject345899AE",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "ABLEdeployProject345899AE",
                    },
                    "-*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ABLEdeployProjectRoleDefaultPolicyF0179D0A",
        "Roles": Array [
          Object {
            "Ref": "ABLEdeployProjectRole33EB96A4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
